src/segment-loader.js

```diff
@@ -1389,53 +1389,61 @@ export default class SegmentLoader extends videojs.EventTarget {
     }
   }
 
-  appendData_(segmentInfo, result) {
-    // Merge multiple video and audio segments into one and append
-    const {
-      type,
-      data
-    } = result;
-    let initSegment = result.initSegment;
+  getInitSegmentAndUpdateState_({ type, initSegment, map, playlist }) {
+    // "The EXT-X-MAP tag specifies how to obtain the Media Initialization Section
+    // (Section 3) required to parse the applicable Media Segments.  It applies to every
+    // Media Segment that appears after it in the Playlist until the next EXT-X-MAP tag
+    // or until the end of the playlist."
+    // https://tools.ietf.org/html/draft-pantos-http-live-streaming-23#section-4.3.2.5
+    if (map) {
+      const id = initSegmentId(map);
+
+      if (this.activeInitSegmentId_ === id) {
+        // don't need to re-append the init segment if the ID matches
+        return null;
+      }
 
-    if (!data || !data.byteLength) {
-      return;
+      // a map-specified init segment takes priority over any transmuxed (or otherwise
+      // obtained) init segment
+      //
+      // this also caches the init segment for later use
+      initSegment = this.initSegmentForMap(map, true).bytes;
+      this.activeInitSegmentId_ = id;
```

Comment:
should this be `initSegment.id`?

Reply:
This is the ID generated by `const id = initSegmentId(map);`

Reply:
are we sure that the initSegmentId generated from the map is the same as the id generated from the init segment generated from the map? In other words is `initSegmentId(map) === this.initSegmentForMap(map, true)`?

Reply:
Those are different values. `initSegmentForMap` returns an init segment object. `initSegmentId` returns an ID string.

Reply:
right, but I mean we're replacing the value of `initSegment` so are we sure that the id doesn't change?

Reply:
initSegment doesn't have an ID property. The ID generated by the map is what is being compared, so as long as we don't modify segment map objects, the IDs should be the same.

Reply:
👍